### PR TITLE
[Finder] make reverse sorting a bit more generic

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/27967#issuecomment-406991194
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Review by fabpot — APPROVED** [review]

File: src/Symfony/Component/Finder/Iterator/ReverseSortingIterator.php [hunk 1]
```diff
@@ -1,32 +0,0 @@
...
- */
-
-namespace Symfony\Component\Finder\Iterator;
-
-/**
- * Reverse the order of a previous iterator.
- *
- * @author Grégoire Pineau <lyrixx@lyrixx.info>
- */
-class ReverseSortingIterator implements \IteratorAggregate
```
Comment (stof) [inline src/Symfony/Component/Finder/Iterator/ReverseSortingIterator.php]: this class should be deprecated, not removed, as it was not marked internal

File: src/Symfony/Component/Finder/Iterator/SortableIterator.php [hunk 2]
```diff
@@ -34,40 +35,43 @@ class SortableIterator implements \IteratorAggregate
...
             };
         } elseif (self::SORT_BY_MODIFIED_TIME === $sort) {
-            $this->sort = function ($a, $b) {
-                return $a->getMTime() - $b->getMTime();
+            $this->sort = function ($a, $b) use ($order) {
+                return $order * ($a->getMTime() - $b->getMTime());
             };
+        } elseif (self::SORT_BY_NONE === $sort) {
+            $this->sort = $order;
         } elseif (\is_callable($sort)) {
```
Comment (stof) [inline src/Symfony/Component/Finder/Iterator/SortableIterator.php]: what about order support for this case ?

File: src/Symfony/Component/Finder/Iterator/ReverseSortingIterator.php (hunk 1 above)
Comment (fabpot) [reply src/Symfony/Component/Finder/Iterator/ReverseSortingIterator.php]: This class only exists in master (as the PR was merged a few hours ago only).

Comment (stof) [reply src/Symfony/Component/Finder/Iterator/ReverseSortingIterator.php]: hmm indeed, I missed that

File: src/Symfony/Component/Finder/Iterator/SortableIterator.php (hunk 2 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Finder/Iterator/SortableIterator.php]: fixed in 4d70a0b7d4

**Pull request merged**